src/Pic.vue

```diff
@@ -17,10 +19,45 @@
         type: String
       },
       height: {
-        type: String
+        type: String,
+        default: ''
       },
       width: {
-        type: String
+        type: String,
+        default: ''
+      }
+    },
+    computed: {
+      hasWidth () {
+        return this.width !== '';
+      },
+      hasHeight () {
+        return this.height !== '';
+      },
+      computedWidth () {
+        if (this.hasWidth) {
+          return this.width;
+        }
+        if (this.hasHeight) {
+          return this.translatedHeight;
+        }
+        return '';
+      }
+    },
+    data () {
+      return {
+        translatedHeight: 0,
+      }
+    },
+    methods: {
+      computeWidth() {
+        if (!this.hasWidth && this.hasHeight) {
+          const renderedImg = this.$refs.pic;
+          const imgHeight = renderedImg.naturalHeight;
+          const imgWidth = renderedImg.naturalWidth;
+          const aspectRatio = imgWidth / imgHeight;
+          this.translatedHeight = Math.round(toNumber(this.height) * aspectRatio);
```

Comment:
This is the ultimately the width though, so it is weird to name it as `translatedHeight`. Maybe just `calculatedWidth`?